os.walkDir breaks if called in a function with a parameter named 'glob'

### Description

```nim
import std/os

when defined(bug):
  proc p(glob: string) =
    for _ in walkFiles(glob): discard
  p("dir/*")
else:
  proc p(g: string) =
    for _ in walkFiles(g): discard
  p("dir/*")
```

### Nim Version

```
Nim Compiler Version 1.6.10 [Linux: amd64]
Compiled at 2022-11-21
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: f1519259f85cbdf2d5ff617c6a5534fcd2ff6942
active boot switches: -d:release
```

and

```
Nim Compiler Version 1.7.3 [Linux: amd64]
Compiled at 2022-12-16
Copyright (c) 2006-2022 by Andreas Rumpf

git hash: 9a50033d5b09ec3263a53bd2bb4182a8a50e6f4d
active boot switches: -d:release
```

### Current Output

```text
$ nim r globname.n
(no issues)

$ nim -d:bug r globname.nim
Hint: used config file '/home/jfondren/.choosenim/toolchains/nim-1.6.10/config/nim.cfg' [Conf]
Hint: used config file '/home/jfondren/.choosenim/toolchains/nim-1.6.10/config/config.nims' [Conf]
......................................................................................
CC: globname.nim
/home/jfondren/.cache/nim/globname_d/@mglobname.nim.c: In function ‘p__globname_2’:
/home/jfondren/.cache/nim/globname_d/@mglobname.nim.c:254:23: error: called object ‘glob’ is not a function or function pointer
  254 |                 T2_ = glob(nimToCStringConv(glob), ((int) 0), NIM_NIL, (&fX60gensym137_));
      |                       ^~~~
/home/jfondren/.cache/nim/globname_d/@mglobname.nim.c:233:61: note: declared here
  233 | N_LIB_PRIVATE N_NIMCALL(void, p__globname_2)(NimStringDesc* glob) {
      |                                              ~~~~~~~~~~~~~~~^~~~
Error: execution of an external compiler program 'gcc -c  -w -fmax-errors=3   -I/home/jfondren/.choosenim/toolchains/nim-1.6.10/lib -I/home/jfondren/nim/bug -o /home/jfondren/.cache/nim/globname_d/@mglobname.nim.c.o /home/jfondren/.cache/nim/globname_d/@mglobname.nim.c' failed with exit code: 1
```

### Expected Output

_No response_

### Possible Solution

Mangle parameter names.

### Additional Information

_No response_

--- a/compiler/ccgtypes.nim
+++ b/compiler/ccgtypes.nim
@@ -48,11 +48,9 @@ proc fillBackendName(m: BModule; s: PSym) =
     writeMangledName(m.ndi, s, m.config)
 
 proc fillParamName(m: BModule; s: PSym) =
-  ## we cannot use 'sigConflicts' here since we have a BModule, not a BProc.
-  ## Fortunately C's scoping rules are sane enough so that that doesn't
-  ## cause any trouble.
   if s.loc.r == "":
     var res = s.name.s.mangle
+    res.add idOrSig(s, res, m.sigConflicts)
     # Take into account if HCR is on because of the following scenario:
     #   if a module gets imported and it has some more importc symbols in it,
     # some param names might receive the "_0" suffix to distinguish from what
@@ -69,8 +67,6 @@ proc fillParamName(m: BModule; s: PSym) =
     # and a function called in main or proxy uses `socket` as a parameter name.
     # That would lead to either needing to reload `proxy` or to overwrite the
     # executable file for the main module, which is running (or both!) -> error.
-    # always mangles the param; bug #21116
-    res.add "_0"
     s.loc.r = res.rope
     writeMangledName(m.ndi, s, m.config)
 

--- a/compiler/ccgtypes.nim
+++ b/compiler/ccgtypes.nim
@@ -69,8 +69,8 @@ proc fillParamName(m: BModule; s: PSym) =
     # and a function called in main or proxy uses `socket` as a parameter name.
     # That would lead to either needing to reload `proxy` or to overwrite the
     # executable file for the main module, which is running (or both!) -> error.
-    if m.hcrOn or isKeyword(s.name) or m.g.config.cppDefines.contains(res):
-      res.add "_0"
+    # always mangles the param; bug #21116
+    res.add "_0"
     s.loc.r = res.rope
     writeMangledName(m.ndi, s, m.config)
 

--- a/tests/ccgbugs/t21116.nim
+++ b/tests/ccgbugs/t21116.nim
@@ -1,0 +1,11 @@
+discard """
+  target: "c cpp"
+  disabled: windows
+"""
+# bug #21116
+import posix
+
+proc p(glob: string) =
+  discard posix.glob(glob, 0, nil, nil)
+
+p "*"

--- a/tests/ccgbugs/t21116.nim
+++ b/tests/ccgbugs/t21116.nim
@@ -3,9 +3,8 @@ discard """
   disabled: windows
 """
 # bug #21116
-import posix
+import std/os
 
 proc p(glob: string) =
-  discard posix.glob(glob, 0, nil, nil)
-
-p "*"
+  for _ in walkFiles(glob): discard
+p("dir/*")

--- a/tests/ccgbugs/tnoalias.nim
+++ b/tests/ccgbugs/tnoalias.nim
@@ -1,5 +1,5 @@
 discard """
-  ccodecheck: "\\i@'NI* NIM_NOALIAS field;' @'NIM_CHAR* NIM_NOALIAS x,' @'void* NIM_NOALIAS q'"
+  ccodecheck: "\\i@'NI* NIM_NOALIAS field;' @'NIM_CHAR* NIM_NOALIAS x__0qEngDE9aYoYsF8tWnyPacw,' @'void* NIM_NOALIAS q'"
 """
 
 type

